src/Benchmarks/MemoryStreamCopy/MemoryStreamCopyBenchmark.cs

```diff
@@ -0,0 +1,176 @@
+﻿using System.Text;
+
+using BenchmarkDotNet.Attributes;
+
+using Benchmarks.Mocks;
+
+using Microsoft.IO;
+
+namespace Benchmarks.Deserialization;
+[ShortRunJob]
+[MemoryDiagnoser]
+#pragma warning disable CA1001 // Types that own disposable fields should be disposable
+public class MemoryStreamCopyBenchmark
+#pragma warning restore CA1001 // Types that own disposable fields should be disposable
+{
+    private MemoryStream? _memoryStream;
+    private CancellationTokenSource? _cancellationTokenSource;
+    private static RecyclableMemoryStreamManager manager = new RecyclableMemoryStreamManager();
+    [GlobalSetup]
+    public void Setup()
+    {
+        _memoryStream = new MemoryStream(Encoding.UTF8.GetBytes(FunTranslationClient.ReturnJson));
+        _cancellationTokenSource = new();
+    }
+    
+    public void IterationSetup()
+    {
+        _memoryStream!.Seek(0, SeekOrigin.Begin);
+    }
+
+    [Benchmark]
+    public async Task<MemoryStream> SingleMemoryCopyAsync()
+    {
+        IterationSetup();
+        using var memoryStream = new MemoryStream();
+        await _memoryStream!.CopyToAsync(memoryStream).ConfigureAwait(false);
+        return memoryStream;
+    }
+
+
+    [Benchmark]
+    public async Task<MemoryStream> SingleRecycableMemoryCopyAsync()
+    {
+        IterationSetup();
+        using var memoryStream = manager.GetStream();
+        await _memoryStream!.CopyToAsync(memoryStream).ConfigureAwait(false);
+        return memoryStream;
+    }
+
+    [Benchmark]
+    public async Task HundredMemoryCopyAsync()
+    {
+        IterationSetup();        
+        for (var i = 0; i < 100; i++) {
+            using (var memoryStream = new MemoryStream())
+            {
+                await _memoryStream!.CopyToAsync(memoryStream).ConfigureAwait(false);
+            }
+        }
+        return;
+    }
+
+
+    [Benchmark]
+    public async Task HundredRecycableMemoryCopyAsync()
+    {
+        IterationSetup();        
```

Comment:
if we keep this test, I believe the iteration setup needs to be within the loop

Reply:
done and created more different approaches to validate the benchmark results over different file sizes and situations.